nixos/modules/services/networking/newt.nix

```
@@ -6,39 +6,73 @@
 }:
 let
   cfg = config.services.newt;
+  cliType =
+    with lib.types;
+    attrsOf (
+      nullOr (oneOf [
+        bool
+        int
+        float
+        str
+        path
+        (listOf cliType)
+      ])
+    )
+    // {
+      description = "value coercible to CLI argument";
+    };
 in
 {
+  imports = [
+    (lib.mkRenamedOptionModule [ "services" "newt" "id" ] [ "services" "newt" "settings" "id" ])
+    (lib.mkRenamedOptionModule
+      [ "services" "newt" "logLevel" ]
+      [ "services" "newt" "settings" "log-level" ]
+    )
+    (lib.mkRenamedOptionModule
+      [ "services" "newt" "endpoint" ]
+      [ "services" "newt" "settings" "endpoint" ]
+    )
+  ];
+
   options = {
     services.newt = {
       enable = lib.mkEnableOption "Newt, user space tunnel client for Pangolin";
-      # needs to be changed when newt-go changes to fosrl-newt
-      package = lib.mkPackageOption pkgs "newt-go" { };
-
-      id = lib.mkOption {
-        type = with lib.types; nullOr str;
-        default = null;
-        description = ''
-          The Newt Id that will be used to communicate to Pangolin. This is generated on site creation in the dashboard.
-        '';
-      };
-      endpoint = lib.mkOption {
-        type = with lib.types; nullOr str;
-        default = null;
-        description = ''
-          The endpoint where both Gerbil and Pangolin reside in order to connect to the websocket. The url of your Pangolin dashboard.
-        '';
-      };
-      logLevel = lib.mkOption {
-        type = lib.types.enum [
-          "DEBUG"
-          "INFO"
-          "WARN"
-          "ERROR"
-          "FATAL"
-        ];
-        default = "INFO";
-        description = "The log level to use.";
+      package = lib.mkPackageOption pkgs "fosrl-newt" { };
+      settings = lib.mkOption {
+        type = lib.types.submodule {
+          freeformType = lib.types.attrsOf cliType;
+          options = {
+            id = lib.mkOption {
+              type = with lib.types; nullOr str;
+              default = null;
+              description = ''
+                The Newt Id that will be used to communicate to Pangolin. This is generated on site creation in the dashboard.
+              '';
+            };
+            endpoint = lib.mkOption {
+              type = with lib.types; nullOr str;
+              default = null;
+              description = ''
+                The endpoint where both Gerbil and Pangolin reside in order to connect to the websocket. The url of your Pangolin dashboard.
+              '';
+            };
+            log-level = lib.mkOption {
+              type = lib.types.enum [
+                "DEBUG"
+                "INFO"
+                "WARN"
+                "ERROR"
+                "FATAL"
+              ];
+              default = "INFO";
+              description = "The log level to use.";
+            };
+          };
+        };
```

```suggestion
        type = cliType;
```
We don't need the value of those options, so a submodule is currently unnecessary.

`attrsOf` is also wrong here, since it's already defined in the `cliType` definition.

Could even rename `cliType` to `type` and use `inherit` here if you fancy that.